File: app/src/main/java/com/robotnec/budget/core/persistence/dao/TransactionDao.java
package com.robotnec.budget.core.persistence.dao;

import com.robotnec.budget.core.domain.operation.Transaction;

import java.util.List;

/**
 * @author zak <zak@swingpulse.com>
 */
public interface TransactionDao extends BaseDao<Transaction> {
    List<Transaction> getTransactionsForCategory(long categoryId);
}

File: app/src/main/java/com/robotnec/budget/core/persistence/dao/TransactionDao.kt
package com.robotnec.budget.core.persistence.dao

import com.robotnec.budget.core.domain.operation.Transaction

/**
 * @author zak zak@swingpulse.com>
 */
interface TransactionDao : BaseDao<Transaction> {
    fun getTransactionsForCategory(categoryId: Long): List<Transaction>
}
